crates/core/src/hash.rs

```diff
@@ -156,6 +157,23 @@ impl Hash {
     }
 }
 
+impl<'a> Add<&'a Hash> for Hash {
```

Comment:
:joy: this is some proper magic shit. I'm somewhat opposed to this `Add` impl, especially because it is not too clear what should be happening when adding two hashes

Reply:
I also wondered if this would be confusing, but then I tried to think what other operations people would guess it to be and I couldn't come up with any.

Reply:
I googled adding hashes and realized xor-ing hashes is incredibly common. So yeah, this is confusing